### DISCUSSION TITLE
AWS SSO in ~/.aws/config

### DISCUSSION THREAD

**Discussion by churnd** [iterate-ch/cyberduck#14529] [Q&A]
AWS SSO seems to save session information in ~/.aws/config, not ~/.aws/credentials:  https://docs.aws.amazon.com/cli/latest/userguide/sso-configure-profile-token.html.  Yet the documentation mentions using ~/.aws/credentials:  https://docs.cyberduck.io/protocols/s3/#connecting-using-credentials-from-aws-command-line-interface.  Which is the right way to do it?  I can't find any information about how to use ~/.aws/credentials.

**Comment by dkocher** [comment]
We read settings from both configuration files [^1], but settings in `~/.aws/config` might override settings in `~/.aws/credentials` [^2]. This relates to #13377.

[^1]: https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html
[^2]: https://github.com/iterate-ch/cyberduck/blob/eff427c6afca57432250e8813e6aed9b31c7750f/s3/src/main/java/ch/cyberduck/core/sts/STSCredentialsConfigurator.java#L100-L114
